File: Design Patterns/Command Pattern/Java/com.faculdadevincit.designpatterns.commandpattern/src/ControleRemoto.java
public class ControleRemoto {
    private ICommand command;

    public ICommand getCommand() {
        return command;
    }

    public void setCommand(ICommand command) {
        this.command = command;
    }

    public void pressionarTecla() {
        command.executar();
    }

    public void desfazerPressionar() {
        command.desfazer();
    }
}

File: Design Patterns/Command Pattern/CSharp/FaculdadeVincit.DesignPatterns.CommandPattern/ControleRemoto.cs
namespace FaculdadeVincit.DesignPatterns.CommandPattern;

public class ControleRemoto
{
    public ICommand? command { get; set; }

    public void PressionarBotao() => command?.Executar();
    public void DesfazerPressionar() => command?.Desfazer();
}
